src/MIDebugEngine/AD7.Impl/AD7Thread.cs

```diff
@@ -276,26 +276,19 @@ int IDebugThread2.Resume(out uint suspendCount)
         // Sets the next statement to the given stack frame and code context.
         int IDebugThread2.SetNextStatement(IDebugStackFrame2 stackFrame, IDebugCodeContext2 codeContext)
         {
-            // CLRDBG TODO: This implementation should be changed to call an MI command
-            ulong addr = ((AD7MemoryAddress)codeContext).Address;
-            AD7StackFrame frame = ((AD7StackFrame)stackFrame);
-            if (frame.ThreadContext.Level != 0 || frame.Thread != this || !frame.ThreadContext.pc.HasValue || _engine.DebuggedProcess.MICommandFactory.Mode == MIMode.Clrdbg)
```

Comment:
The first two parts of this `if` (checking that the level is zero, and that the frame is for the correct thread) still make sense if a non-null stack frame is provided.

Reply:
According to the docs this is reserved and should be null.

Reply:
The docs are wrong. VS will always provide the frame. The frame is pretty useless, but it is provided.

Reply:
Should be fixed then.
I wonder why they added that parameter.

Reply:
All of the '2' interfaces in AD7 (the name for the IDebug* family of interfaces) date back to the early 2000s and many are older than that (because they came from a previous product in the Visual Studio family and were redone a bit for VS 2002). So the reason has been lost to time. But I would **guess** either:
1. The designers wanted to support engines that could automatically unwind out of frames. This doesn't really make any sense for a native debuggger. But managed languages might know enough to be able to do that.
-or-
2. It was just an oversight